src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest2.cs

```diff
@@ -105,14 +105,30 @@ public void CompressedWithoutLinker ()
 		}
 
 		[Test]
-		public void BuildBasicApplication ([ValueSource (nameof (SupportedTargetFrameworks))] string tfv, [Values (true, false)] bool isRelease)
+		public void BuildBasicApplication ([ValueSource (nameof (SupportedTargetFrameworks))] string tfv, [Values (true, false)] bool isRelease, [Values ("", "en_US.UTF-8", "sv_SE.UTF-8")] string langEnvironmentVariable)
 		{
-			var proj = new XamarinAndroidApplicationProject {
-				IsRelease = isRelease,
-				TargetFrameworkVersion = tfv,
-			};
-			using (var b = CreateApkBuilder ()) {
-				Assert.IsTrue (b.Build (proj), "Build should have succeeded.");
+			string? oldLangVar = null;
+
+			if (!String.IsNullOrEmpty (langEnvironmentVariable)) {
+				// See: https://github.com/xamarin/xamarin-android/pull/7941
+				oldLangVar = Environment.GetEnvironmentVariable ("LANG");
+
+				// This will **not** call `setenv(3)`, but it should be fine since we're after the managed code here, not native
+				Environment.SetEnvironmentVariable ("LANG", langEnvironmentVariable);
```

Comment:
I'm concerned about "thread safety" here; we're modifying global state here, so this test needs to be run "alone" lest we anger the deity of randomness.  Unfortunately, I don't think we're safe, as this fixture is `[Parallelizable (ParallelScope.Children)]`.

Perhaps all we need to do is add `[NonParallelizable]` to this test method?